test/media-groups.test.js

```diff
@@ -18,1208 +18,1426 @@ QUnit.module('MediaGroups', {
+        { id: 'fr', playlistLoader: mockPlaylistLoader() },
+        { id: 'es', playlistLoader: mockPlaylistLoader() }
+      ];
+      tracks.en = { id: 'en', enabled: false };
+      tracks.fr = { id: 'fr', enabled: false };
+      tracks.es = { id: 'es', enabled: false };
+      mediaType.activeTrack = MediaGroups.activeTrack[type](type, settings);
+      mediaType.activeGroup = MediaGroups.activeGroup(type, settings);
+      mediaType.getActiveGroup = MediaGroups.getActiveGroup(type, settings);
+
+      const onTrackChanged = MediaGroups.onTrackChanged(type, settings);
+
+      onTrackChanged();
+
+      assert.equal(segmentLoaderPauseCalls, 1, 'loaders paused on track change');
+      assert.equal(mainSegmentLoaderResetCalls, 0, 'no main reset when no active group');
+      assert.equal(mainSegmentLoaderSetAudioCalls.length, 0, 'no main setAudio when no active group');
+      assert.equal(segmentLoaderResetCalls, 0, 'no reset when no active group');
+      assert.equal(segmentLoaderSetAudioCalls.length, 0, 'no setAudio when no active group');
+
+      tracks.en.enabled = true;
+
+      onTrackChanged();
+
+      assert.equal(segmentLoaderPauseCalls, 2, 'loaders paused on track change');
+      assert.equal(
+        mainSegmentLoaderResetCalls, 1,
+        'main reset when changing to group with no playlist loader'
+      );
+      assert.equal(
+        mainSegmentLoaderSetAudioCalls.length, 1,
+        'main audio set when changing to group with no playlist loader'
+      );
+      assert.ok(mainSegmentLoaderSetAudioCalls[0], 'main audio set to true');
+      assert.equal(
+        segmentLoaderResetCalls, 0,
+        'no reset changing to group with no playlist loader'
+      );
+      assert.equal(
+        segmentLoaderSetAudioCalls.length, 0,
+        'no audio set when changing to group with no playlist loader'
+      );
+
+      tracks.en.enabled = false;
+      tracks.fr.enabled = true;
+      mediaType.activePlaylistLoader = groups.main[1].playlistLoader;
+
+      onTrackChanged();
+
+      assert.equal(segmentLoaderPauseCalls, 3, 'loaders paused on track change');
+      assert.equal(
+        mainSegmentLoaderResetCalls, 1,
+        'no main reset changing to group with playlist loader'
+      );
+      assert.equal(
+        mainSegmentLoaderSetAudioCalls.length, 2,
+        'main audio set when changing to group with playlist loader'
+      );
+      assert.notOk(mainSegmentLoaderSetAudioCalls[1], 'main audio set to true');
+      assert.equal(
+        segmentLoaderResetCalls, 0,
+        'no reset when active group hasn\'t changed'
+      );
+      assert.equal(segmentLoaderSetAudioCalls.length, 1, 'set audio on track change');
+      assert.ok(segmentLoaderSetAudioCalls[0], 'enabled audio on track change');
+      assert.strictEqual(
+        mediaType.activePlaylistLoader, groups.main[1].playlistLoader,
+        'sets the correct active playlist loader'
+      );
+
+      tracks.fr.enabled = false;
```

Comment:
This test previously relied on weird logic to change tracks, now we actually change the tracks around before each step.